### PR TITLE
expr: concrete function definition for `pg_get_constraintref`

### PR DESCRIPTION
**Pull request by petrosagg**
### Motivation

Apart from the possibility of implementing it in the future, this aligns the function with the effort of migrating all functions away from the raw enum definition style.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.

### CODE REVIEW

File: src/expr/src/scalar/func.rs [hunk 1]
```diff
@@ -3217,6 +3233,7 @@ pub enum UnaryFunc {
     Sleep(Sleep),
     RescaleNumeric(u8),
     PgColumnSize(PgColumnSize),
+    PgGetConstraintdef(PgGetConstraintdef),
```
Comment (guswynn) [inline src/expr/src/scalar/func.rs:3236]: This is very cool, didn't know that the new way let you share impls with different numbers of args so easily!

**Review by guswynn — APPROVED** [review]
thanks for cleaning this up!

**Review by sploiselle — APPROVED** [review]

**Comment by benesch** [comment]
Wait, why is this even getting all the way to dataflow?

**Comment by benesch** [comment]
Oh, I see, so that as long as it gets optimized away it's fine. I don't think the error message "pg_get_constraintdef with non-null arguments" is quite right; even `NULL` arguments will fail to evaluate. It's just that in some cases the evaluation will get optimized away.

**Comment by petrosagg** [comment]
@benesch I see, I copied the behaviour that got merged in main. If you try a `SELECT pg_get_constraintdef(NULL);` on main it produces an answer just fine. I'll change this PR to always error out

**Review by benesch — APPROVED** [review]

**Comment by benesch** [comment]
Not sure why the `CARGO_INCREMENTAL=0` isn't seeming to help the macOS build. https://github.com/rust-lang/rust/pull/91919 is close to merging, though.

**Comment by benesch** [comment]
I got frustrated and busted the cache on the macOS builder agent manually. It seems like the underlying bug is likely to get fixed soon in rustc, and I'm happy to periodically manually bust the cache as necessary until that fix ships. So @petrosagg merge away!

**Comment by benesch** [comment]
I'm going to merge this in the hope that the `CARGO_INCREMENTAL=0` thing helps out @andrioni in #9682.

**Comment by petrosagg** [comment]
@benesch if it's for the mac builder, it doesn't. I don't know why it is not picked up when set in the yaml file. When I put the same flag in `./bin/check` itself the mac builder correctly built my branch. It might be some shortcoming of builtkite on mac agents? Not sure

**Pull request merged**